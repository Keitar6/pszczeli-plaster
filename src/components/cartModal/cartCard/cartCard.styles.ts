import { Colors, Typhography } from "../../../global.styles";
import styled from "styled-components";

export const CartItemContainer = styled.div`
  width: 100%;
  display: flex;
  height: 5rem;
  //   margin-bottom: 1rem;
  padding: 0.5rem 0;
  border-bottom: 1px solid ${Colors.darkRGBA};
`;
export const Image = styled.img`
  width: 7rem;
  padding-right: 0.5rem;
  border-right: 1px solid ${Colors.darkRGBA};
`;

export const ItemDetails = styled.div`
  width: 80%;
  flex-basis: auto;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 0rem 1rem;
  gap: 1rem;
`;

export const Name = styled.span`
  width: 100%;
  font-size: ${Typhography.fontSizes.H6};
  //   border: 2px solid red;
`;

export const Price = styled.span`
  display: flex;
  width: 100%;
  font-size: ${Typhography.fontSizes.H6};
  font-weight: bold;
  //   border: 2px solid red;
`;
