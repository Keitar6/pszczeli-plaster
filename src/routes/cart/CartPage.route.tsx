export const CartPage = () => {
  return (
    <h2>
      CartPage PAGE
    </h2>
  );
};