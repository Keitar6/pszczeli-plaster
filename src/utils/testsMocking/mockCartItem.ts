import { CategoryItem } from "../../store/categories/category.types";

export const mockCartItemInArray: CategoryItem & { quantity: number } = {
  id: "ajdi",
  name: "name",
  weight: 205,
  price: 13,
  image: "yes",
  dodatki: false,
  quantity: 2
};

export const mockCartItemInArray1Quantity: CategoryItem & { quantity: number } =
  {
    id: "ajdi",
    name: "name",
    weight: 205,
    price: 13,
    image: "yes",
    dodatki: false,
    quantity: 1
  };

export const mockCartItemNotInArray: CategoryItem & { quantity: number } = {
  id: "ajdi",
  name: "name",
  weight: 205,
  price: 13,
  image: "yes",
  dodatki: false,
  quantity: 0
};

export const mockCartItems = [
  {
    id: "FajneID",
    name: "SuperName",
    weight: 15,
    price: 165,
    image: "ImageSciezka",
    dodatki: false,
    quantity: 2
  },
];

export const mockCartItems2 = [
  {
    id: "FajneID",
    name: "SuperName",
    weight: 15,
    price: 165,
    image: "ImageSciezka",
    dodatki: false,
    quantity: 2
  },
  {
    id: "AjneID",
    name: "UperName",
    weight: 14,
    price: 42,
    image: "ImageSciezka",
    dodatki: false,
    quantity: 1
  }
];