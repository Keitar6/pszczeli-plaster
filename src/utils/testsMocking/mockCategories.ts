export const mockCategoryItem = {
  id: "001M",
  name: "Miód lipowy",
  dodatki: false,
  image: "Miody/lipa.jpg",
  weight: 500,
  price: 25
};
export const mockCategoryItemArray = [
  {
    id: "001M",
    name: "Miód lipowy",
    dodatki: false,
    image: "Miody/lipa.jpg",
    weight: 500,
    price: 25
  },
  {
    id: "002M",
    name: "Miód rzepakowy",
    dodatki: false,
    image: "Miody/rzepak.jpg",
    weight: 500,
    price: 24
  },
  {
    id: "003M",
    name: "Miód wielokwiatowy",
    dodatki: false,
    image: "Miody/wielokwiat.jpg",
    weight: 500,
    price: 20
  }
];
export const mockCategories = [
  {
    title: "Miody",
    icon: "emojione:honey-pot",
    items: [
      {
        id: "001M",
        name: "Miód lipowy",
        dodatki: false,
        image: "Miody/lipa.jpg",
        weight: 500,
        price: 25
      },
      {
        id: "002M",
        name: "Miód rzepakowy",
        dodatki: false,
        image: "Miody/rzepak.jpg",
        weight: 500,
        price: 24
      },
      {
        id: "003M",
        name: "Miód wielokwiatowy",
        dodatki: false,
        image: "Miody/wielokwiat.jpg",
        weight: 500,
        price: 20
      }
    ]
  },
  {
    title: "Miody pitne",
    icon: "icon-park-solid:honey-one",
    items: [
      {
        id: "001MP",
        name: "Miód pitny trójniak",
        image: "MiodyPitne/rogDoMiodu.jpg",
        weight: 500,
        price: 35
      }
    ]
  },
  {
    title: "Produkty pszczele",
    icon: "noto-v1:honeybee",
    items: [
      {
        id: "001PP",
        name: "Pyłek pszczeli",
        image: "ProduktyPszczele/PylekPszczeli.jpg",
        weight: 500,
        price: 48
      }
    ]
  }
];

export const mockCategoriesMap = {
  miody: [
    {
      id: "001M",
      name: "Miód lipowy",
      dodatki: false,
      image: "Miody/lipa.jpg",
      weight: 500,
      price: 25
    },
    {
      id: "002M",
      name: "Miód rzepakowy",
      dodatki: false,
      image: "Miody/rzepak.jpg",
      weight: 500,
      price: 24
    },
    {
      id: "003M",
      name: "Miód wielokwiatowy",
      dodatki: false,
      image: "Miody/wielokwiat.jpg",
      weight: 500,
      price: 20
    }
  ],

  "miody pitne": [
    {
      id: "001MP",
      name: "Miód pitny trójniak",
      image: "MiodyPitne/rogDoMiodu.jpg",
      weight: 500,
      price: 35
    }
  ],

  "produkty pszczele": [
    {
      id: "001PP",
      name: "Pyłek pszczeli",
      image: "ProduktyPszczele/PylekPszczeli.jpg",
      weight: 500,
      price: 48
    }
  ]
};

export const mockAllItemsMap = [
  {
    id: "001M",
    name: "Miód lipowy",
    dodatki: false,
    image: "Miody/lipa.jpg",
    weight: 500,
    price: 25
  },
  {
    id: "002M",
    name: "Miód rzepakowy",
    dodatki: false,
    image: "Miody/rzepak.jpg",
    weight: 500,
    price: 24
  },
  {
    id: "003M",
    name: "Miód wielokwiatowy",
    dodatki: false,
    image: "Miody/wielokwiat.jpg",
    weight: 500,
    price: 20
  },

  {
    id: "001MP",
    name: "Miód pitny trójniak",
    image: "MiodyPitne/rogDoMiodu.jpg",
    weight: 500,
    price: 35
  },

  {
    id: "001PP",
    name: "Pyłek pszczeli",
    image: "ProduktyPszczele/PylekPszczeli.jpg",
    weight: 500,
    price: 48
  }
];
