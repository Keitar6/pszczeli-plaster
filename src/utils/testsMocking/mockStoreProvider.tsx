import React, { PropsWithChildren } from "react";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import type { PreloadedState } from "@reduxjs/toolkit";

import rootReducer, { ReduxState } from "../../store/rootReducer.redux";
import { Provider } from "react-redux";
import { AppStore } from "../../store/store.redux";
import { BrowserRouter } from "react-router-dom";

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<ReduxState>;
  store?: typeof AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: rootReducer as any, preloadedState }),
    ...renderOptions
  }: ExtendedRenderOptions = {},
  ifWithRouter = true
) {
  function Wrapper({ children }: PropsWithChildren): JSX.Element {
    if (ifWithRouter)
      return (
        <BrowserRouter>
          <Provider store={store}>{children}</Provider>
        </BrowserRouter>
      );
    else return <Provider store={store}>{children}</Provider>;
  }
  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
