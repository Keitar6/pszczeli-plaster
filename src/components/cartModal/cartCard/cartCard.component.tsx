import { CartItem } from "store/cartReducer/cart.types";

import {
  CartItemContainer,
  Image,
  ItemDetails,
  Name,
  Price
} from "./cartCard.styles";

export type CartItemProps = {
  cartItem: CartItem;
};

export const CartCard = ({ cartItem }: CartItemProps) => {
  const { name, quantity, price, image } = cartItem;

  return (
    <CartItemContainer>
      <Image
        src={`/dataBaseImages/${image}`}
        alt={`Obraz: ${name}`}
      />
      <ItemDetails>
        <Name>{name}</Name>
        <Price>{`${quantity} x ${price} = ${quantity * price}z≈Ç`}</Price>
      </ItemDetails>
    </CartItemContainer>
  );
};
