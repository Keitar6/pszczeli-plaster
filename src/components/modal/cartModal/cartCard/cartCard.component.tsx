import { CartItem } from "store/cartReducer/cart.types";

import {
  CartItemContainer,
  Image,
  ItemDetails,
  Name,
  Price
} from "./cartCard.styles";

export type CartItemProps = {
  cartItem: CartItem;
};

const CartMenuCard = ({ cartItem }: CartItemProps) => {
  const { name, quantity, price, image } = cartItem;

  return (
    <CartItemContainer>
      <Image src={image} alt={name} />
      <ItemDetails>
        <Name>{name}</Name>
        <Price className="price">{`${quantity} x ${price} = $${
          quantity * price
        }`}</Price>
      </ItemDetails>
    </CartItemContainer>
  );
};

export default CartMenuCard;
