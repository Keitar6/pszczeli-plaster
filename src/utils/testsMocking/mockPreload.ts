import { mockCartItems2 } from "./mockCartItem";
import { mockOrderHistory } from "./mockOrderHistory";

export const currentProductCardMock = {
  id: "fsda",
  name: "dobre",
  price: 12,
  image: "dobry",
  dodatki: false,
  quantity: 1
};

export const preload = {
  generalPropReducer: {
    path: "sklep",
    isUserMenuOpened: false,
    isSortingInView:false,
    productCardModal: {
      isProductCardOpened: false,
      currentProductCard: currentProductCardMock
    },
    viewLimiter: 1
  },
  cartReducer: {
    cartItems: [...mockCartItems2],
    isDropdownActive: false,
    _persist: { version: 1, rehydrated: true }
  },
  orderHistoryReducer: {
    delivery: {
      price: 9,
      type: "KurierDHL" as const
    },
    orderHistory: mockOrderHistory,
    isLoading: false,
    error: null
  }
};
