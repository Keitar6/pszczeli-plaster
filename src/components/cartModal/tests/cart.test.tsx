import ComponentsRenderer from "react-test-renderer";
import { mockCartItems } from "../../../utils/testsMocking/mockCartItem";
import { CartModal } from "../cart.component";
import { CartCard } from "../cartCard/cartCard.component";

import { renderWithProviders } from "../../../utils/testsMocking/mockStoreProvider";
import { preload } from "../../../utils/testsMocking/mockPreload";

describe("cartModal", () => {
  const component = () =>
    renderWithProviders(
      <>
        <div id="modal"></div>
        <CartModal />
      </>,

      {
        preloadedState: {
          ...preload
        }
      }
    );

  test("cartCard", () => {
    const component = ComponentsRenderer.create(
      <CartCard cartItem={mockCartItems[0]} />
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  test("cartModal close", () => {
    const cartClosingCheck = component().getByTestId("cartClosingCheck");
    const spyOnComponent = jest.spyOn(cartClosingCheck, "click");

    expect(cartClosingCheck).toBeTruthy();
    cartClosingCheck.click();

    expect(spyOnComponent).toHaveBeenCalledTimes(1);
  });

  test("cartModal goToCheckout", () => {
    const goToCheckout = component().getByTestId("goToCheckout");
    expect(goToCheckout).toBeTruthy();
    const spyOnComponent = jest.spyOn(goToCheckout, "click");
    goToCheckout.click();

    expect(spyOnComponent).toHaveBeenCalledTimes(1);
  });
});
