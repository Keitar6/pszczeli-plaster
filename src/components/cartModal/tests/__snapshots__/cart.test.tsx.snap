// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cartModal cartCard 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5rem;
  padding-bottom: 0.5rem;
  border-bottom: 1px solid #0230473e;
}

.c1 {
  width: 25%;
  -webkit-flex-basis: 20%;
  -ms-flex-preferred-size: 20%;
  flex-basis: 20%;
}

.c2 {
  width: 75%;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0rem 1rem;
  gap: 1rem;
}

.c3 {
  width: 100%;
  font-size: 1 rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  font-size: 1 rem;
  font-weight: bold;
}

<div
  className="c0"
>
  <img
    alt="Obraz: SuperName"
    className="c1"
    src="/dataBaseImages/ImageSciezka"
  />
  <div
    className="c2"
  >
    <span
      className="c3"
    >
      SuperName
    </span>
    <span
      className="c4"
    >
      2 x 165 = 330z≈Ç
    </span>
  </div>
</div>
`;
